Add isTabOrSpace to WTF

annevk opened the pull request:
#### 2ae8e53cdb4a77a7b7ebf9c8a790fbe8099f60eb
<pre>
Add isTabOrSpace to WTF
<a href="https://bugs.webkit.org/show_bug.cgi?id=255842">https://bugs.webkit.org/show_bug.cgi?id=255842</a>
rdar://108426795

Reviewed by Darin Adler.

Consolidates a number of separate initiatives at defining (and inlining) tab &apos;\t&apos; or space &apos; &apos;.

* Source/WTF/wtf/ASCIICType.h:
(WTF::isASCIITabOrSpace):
* Source/WebCore/Modules/fetch/FetchBodyConsumer.cpp:
(WebCore::parseParameters):
* Source/WebCore/html/track/VTTCue.cpp:
(WebCore::VTTCue::setCueSettings):
* Source/WebCore/html/track/VTTRegion.cpp:
(WebCore::VTTRegion::setRegionSettings):
* Source/WebCore/html/track/WebVTTParser.cpp:
(WebCore::WebVTTParser::checkAndCreateRegion):
(WebCore::WebVTTParser::checkStyleSheet):

The comment here talks about tab or space, but this preserves the existing functionality for now,
modulo using isASCIIWhitespace instead of something locally defined. This might be worth following
up on.

* Source/WebCore/html/track/WebVTTParser.h:
* Source/WebCore/loader/HTTPHeaderField.cpp:
(WebCore::HTTPHeaderField::create):
* Source/WebCore/loader/HeaderFieldTokenizer.cpp:
(WebCore::HeaderFieldTokenizer::consume):
(WebCore::HeaderFieldTokenizer::skipSpaces):
* Source/WebCore/loader/LinkHeader.cpp:
(WebCore::isParameterValueChar):
(WebCore::findURLBoundaries):
(WebCore::parseParameterDelimiter):
(WebCore::parseParameterName):
(WebCore::parseParameterValue):
(WebCore::isSpaceOrTab): Deleted.
* Source/WebCore/platform/network/HTTPParsers.cpp:
(WebCore::skipWhiteSpace):
(WebCore::skipValue):
(WebCore::isValidHTTPHeaderValue):
(WebCore::extractMIMETypeFromMediaType):
(WebCore::parseRange):
(WebCore::isTabOrSpace): Deleted.
* Source/WebCore/platform/network/RFC7230.cpp:
(RFC7230::isQuotedTextCharacter):
(RFC7230::isQuotedPairSecondOctet):
(RFC7230::isCommentText):
(RFC7230::isValidValue):
* Source/WebCore/platform/network/RFC7230.h:
(RFC7230::isWhitespace): Deleted.
* Source/WebCore/platform/network/RFC8941.cpp:
(RFC8941::parseDictionary):

Canonical link: <a href="https://commits.webkit.org/263354@main">https://commits.webkit.org/263354@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c72ceeb538572c712d61f6683f9908e9abc447cd

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/4338 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/4456 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/4584 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/5812 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/4557 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/4330 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/4571 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/4421 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/4783 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/4398 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/4549 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/3904 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/5811 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/2045 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/3881 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/7179 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/3602 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/3875 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/3948 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/5482 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/4122 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/4355 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/3538 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/4436 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/3874 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/3879 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/1104 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/1074 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/7930 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/4537 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/4228 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/1211 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/loader/HTTPHeaderField.cpp:
```diff
@@ -32,8 +32,8 @@ namespace WebCore {
 {
-    StringView strippedName = StringView(unparsedName).stripLeadingAndTrailingMatchedCharacters(RFC7230::isWhitespace<UChar>);
-    StringView strippedValue = StringView(unparsedValue).stripLeadingAndTrailingMatchedCharacters(RFC7230::isWhitespace<UChar>);
+    StringView strippedName = StringView(unparsedName).stripLeadingAndTrailingMatchedCharacters(isASCIITabOrSpace<UChar>);
```
It's unfortunate it can't find UChar on its own, but it's beyond my expertise to address that.

darinadler reviewed (approved):
This refactoring looks good. I am not 100% sure what the ASCII in this function name means. Since it checks for only two characters it is correct for ASCII, UTF-8, UTF-16, Unicode characters, whatever. Many other ASCII functions are named as such because they are not correct for all non-ASCII characters and so it’s important to express in their names that they return false for such cases,

annevk committed:
Add isTabOrSpace to WTF
https://bugs.webkit.org/show_bug.cgi?id=255842
rdar://108426795

Reviewed by Darin Adler.

Consolidates a number of separate initiatives at defining (and inlining) tab '\t' or space ' '.

* Source/WTF/wtf/ASCIICType.h:
(WTF::isASCIITabOrSpace):
* Source/WebCore/Modules/fetch/FetchBodyConsumer.cpp:
(WebCore::parseParameters):
* Source/WebCore/html/track/VTTCue.cpp:
(WebCore::VTTCue::setCueSettings):
* Source/WebCore/html/track/VTTRegion.cpp:
(WebCore::VTTRegion::setRegionSettings):
* Source/WebCore/html/track/WebVTTParser.cpp:
(WebCore::WebVTTParser::checkAndCreateRegion):
(WebCore::WebVTTParser::checkStyleSheet):

The comment here talks about tab or space, but this preserves the existing functionality for now,
modulo using isASCIIWhitespace instead of something locally defined. This might be worth following
up on.

* Source/WebCore/html/track/WebVTTParser.h:
* Source/WebCore/loader/HTTPHeaderField.cpp:
(WebCore::HTTPHeaderField::create):
* Source/WebCore/loader/HeaderFieldTokenizer.cpp:
(WebCore::HeaderFieldTokenizer::consume):
(WebCore::HeaderFieldTokenizer::skipSpaces):
* Source/WebCore/loader/LinkHeader.cpp:
(WebCore::isParameterValueChar):
(WebCore::findURLBoundaries):
(WebCore::parseParameterDelimiter):
(WebCore::parseParameterName):
(WebCore::parseParameterValue):
(WebCore::isSpaceOrTab): Deleted.
* Source/WebCore/platform/network/HTTPParsers.cpp:
(WebCore::skipWhiteSpace):
(WebCore::skipValue):
(WebCore::isValidHTTPHeaderValue):
(WebCore::extractMIMETypeFromMediaType):
(WebCore::parseRange):
(WebCore::isTabOrSpace): Deleted.
* Source/WebCore/platform/network/RFC7230.cpp:
(RFC7230::isQuotedTextCharacter):
(RFC7230::isQuotedPairSecondOctet):
(RFC7230::isCommentText):
(RFC7230::isValidValue):
* Source/WebCore/platform/network/RFC7230.h:
(RFC7230::isWhitespace): Deleted.
* Source/WebCore/platform/network/RFC8941.cpp:
(RFC8941::parseDictionary):

Canonical link: https://commits.webkit.org/263354@main
